src/main/java/com/untamedears/ItemExchange/utility/PotionUtility.java

```
@@ -0,0 +1,293 @@
+			if (upgraded) return "Potion of Poison II";
+			else return "Potion of Poison";
+		}
+		// Regeneration
+		else if (potionType.equals(PotionType.REGEN)) {
+			if (upgraded) return "Potion of Regeneration II";
+			else return "Potion of Regeneration";
+		}
+		// Strength
+		else if (potionType.equals(PotionType.STRENGTH)) {
+			if (upgraded) return "Potion of Strength II";
+			else return "Potion of Strength";
+		}
+		// Weakness
+		else if (potionType.equals(PotionType.WEAKNESS)) {
+			return "Potion of Weakness";
+		}
+		// Luck
+		else if (potionType.equals(PotionType.LUCK)) {
+			return "Potion of Luck";
+		}
+		// NONE OF THE ABOVE
+		else {
+			return "Unknown Potion";
+		}
+	}
+
+	public static PotDurations getVanillaEffectLength(PotionType potion) {
+		// If not a vanilla potion, return -1
+		if (!VANILLA_EFFECTS.contains(potion)) return new PotDurations(-1);
+		// Get the effect type
+		PotionEffectType effectType = potion.getEffectType();
+		// If the effect is instant, return 0
+		if (effectType.isInstant()) return new PotDurations(0);
+		// Go through
+		switch (potion) {
+			case NIGHT_VISION: return new PotDurations(180, 480, -1);
+			case INVISIBILITY: return new PotDurations(180, 480, -1);
+			case JUMP: return new PotDurations(180, 480, 90);
+			case FIRE_RESISTANCE: return new PotDurations(180, 480, -1);
+			case SPEED: return new PotDurations(180, 480, 90);
+			case SLOWNESS: return new PotDurations(90, 240, -1);
+			case WATER_BREATHING: return new PotDurations(180, 480, -1);
+			case POISON: return new PotDurations(45, 90, 21);
+			case REGEN: return new PotDurations(45, 90, 21);
+			case STRENGTH: return new PotDurations(180, 480, 90);
+			case WEAKNESS: return new PotDurations(90, 240, -1);
+			case LUCK: return new PotDurations(300, -1, -1);
+			default: return new PotDurations(-1);
+		}
+	}
+
+	public static String colourToString(Color colour) {
+		try {
+			return Integer.toHexString(colour.asRGB());
+		}
+		catch (Exception e) {
```

Very bad habit to silently catch Exception

---

This is also being done in a bunch of other places